File: desktop-client/src/app/game/services/prize/prize.service.ts
import { Injectable } from '@angular/core';
import { WebsocketService } from '@app/shared/services/websocket/websocket.service';
import { PlayerPrize } from '@common/interfaces/player-prize';
import { GameEvent } from '@common/socket-events/game-event';

@Injectable({
    providedIn: 'root',
})
export class PrizeService {
    gamePrize: number = 0;
    potPrize: number = 0;
    constructor(private readonly websocketService: WebsocketService) {}

    initialize(): void {
        this.websocketService.on<PlayerPrize>(GameEvent.MONEY_PRIZE, (prize) => {
            if (!prize) return;
            this.gamePrize = prize.gamePrize;
            this.potPrize = prize.potPrize;
        });
    }

    removeListeners(): void {
        this.websocketService.removeAllListeners(GameEvent.MONEY_PRIZE);
        this.gamePrize = 0;
        this.potPrize = 0;
    }
}

File: mobile_client/lib/services/game/prize_service.dart
import 'package:flutter/foundation.dart';
import 'package:mobile_client/constants/socket_events/game_event.dart';
import 'package:mobile_client/models/game/player_prize.dart';
import 'package:mobile_client/services/websocket_service.dart';

class PrizeService extends ChangeNotifier {
  PrizeService(this.webSocketService);

  int gamePrize = 0;
  int potPrize = 0;

  final WebSocketService webSocketService;

  void initialize() {
    webSocketService.on<Map<String, dynamic>>(GameEvent.moneyPrize.name,
        (data) {
      final prize = PlayerPrize.fromJson(data);
      gamePrize = prize.gamePrize;
      potPrize = prize.potPrize;
      notifyListeners();
    });
  }

  void removeListeners() {
    webSocketService.removeAllListeners(GameEvent.moneyPrize.name);
    gamePrize = 0;
    potPrize = 0;
    notifyListeners();
  }
}
